app/plugin.go

```diff
@@ -312,6 +314,18 @@ func (a *App) EnablePlugin(id string) *model.AppError {
 		return model.NewAppError("EnablePlugin", "app.plugin.not_installed.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	isCompatible, err := utils.IsPluginManifestCompatibleWithConfig(manifest, a.Config())
```

Comment:
I'm wondering if it makes sense to rework this check as something akin to the plugin helpers, and have plugins opt-in by running these checks `OnActivate`. 

The primary benefit would be to expose this config-checking code to plugins running even on older Mattermost versions. The secondary benefit is that we push some of this complexity downstream, allowing us to iterate on changes without having to constantly evaluate backwards compatibility. (e.g. the plugin has to update its go-modules to get a new version of the helpers to get the new behaviour, otherwise they can always rely on what they have as being stable).

In an ideal world, we might have checked the minimum server version this way too, making it backwards compatible prior to before that check was introduced in the server.

Strawman proposal:
* keep all the `utils.*` helper stuff, but remove the code from this file/api/etc.
* move `IsPluginCompatible` inside the `plugins` package, maybe as `func CheckRequiredConfig(requiredConfig, actualConfig *model.Config)`
* prototype changes in mattermost-plugin-starter-template
* `bin/manifest` in the plugin tooling could extract the `RequiredConfig` from the `plugin.json` (at compile time) and update the generated `manifest.go`
* add an `activate.go` with a pre-defined `OnActivate` implementation that invokes `CheckRequiredConfig`. New plugins could extend this hook as needed, or remove the check if they didn't want to opt-in. Old plugins could use this repository as an example.

Thoughts?

Reply:
makes perfect sense! I misunderstood you yesterday then, I thought we'd be exposing the method to the plugin so that they were able to do the checks when the configuration changed.

Taking a stab at it.

Reply:
In order for the plugins to have access to `CheckRequiredConfig` it should be exposed on the plugin API, right? on `plugin/api.go`